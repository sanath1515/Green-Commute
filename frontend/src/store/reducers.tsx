import { optionsType } from "../Constants";

export type greenState = {
  location: string;
  worklocations: string[];
  skills: string[];
  step: string;
  jobId: string;
  searchSkill: string;
  searchLocation: string;
  options: optionsType;
  optionValue: boolean;
  sidebarIndex: string;
};

export const initialGreenState: greenState = {
  location: "",
  worklocations: [],
  skills: ["UI/UX Designer"],
  step: "1",
  jobId: "1",
  searchSkill: "",
  searchLocation: "",
  options: {
    datePosted: [],
    distance: [],
    experienceLevel: [],
    jobType: [],
    transport: [],
  },
  optionValue: true,
  sidebarIndex: "1",
};

type Action = {
  type: string;
  payload: string | string[] | any | boolean;
};

export const greenReducer = (
  state: greenState = initialGreenState,
  action: Action
) => {
  switch (action.type) {
    case "SET_LOCATION": {
      return { ...state, location: action.payload };
    }
    case "SET_SKILLS": {
      return { ...state, skills: action.payload };
    }
    case "SET_STEP": {
      return { ...state, step: action.payload };
    }
    case "SET_WORK_LOCATIONS": {
      return { ...state, worklocations: action.payload };
    }
    case "SET_JOB_ID": {
      return { ...state, jobId: action.payload };
    }
    case "SET_SEARCH_SKILL": {
      return { ...state, searchSkill: action.payload };
    }
    case "SET_SEARCH_LOCATION": {
      return { ...state, searchLocation: action.payload };
    }
    case "SET_OPTIONS": {
      return { ...state, options: action.payload };
    }
    case "SET_OPTIONS_VALUE": {
      return { ...state, optionValue: action.payload };
    }
    case "SET_SIDEBAR_INDEX": {
      return { ...state, sidebarIndex: action.payload };
    }
    default:
      return state;
  }
};
