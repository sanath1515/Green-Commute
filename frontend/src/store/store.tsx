import { createStore } from "redux";
import { greenReducer } from "./reducers";

const store = createStore(greenReducer);

export default store;
