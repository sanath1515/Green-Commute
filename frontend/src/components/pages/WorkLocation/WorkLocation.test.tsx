/**
 * @jest-environment jsdom
 */

import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import store from "../../../store/store";
import WorkLocation from "./WorkLocation";

describe("WorkLocation Page Test", () => {
  const page = () => {
    render(
      <Router>
        <Provider store={store}>
          <WorkLocation status={0} />
        </Provider>
      </Router>
    );
  };

  it("Match Snapshot", () => {
    expect(page).toMatchSnapshot();
  });

  it("To be Required", () => {
    const container = () =>
      render(
        <Router>
          <Provider store={store}>
            <WorkLocation status={0} />
          </Provider>
        </Router>
      );
    expect(container).toBeRequired;
  });
});
