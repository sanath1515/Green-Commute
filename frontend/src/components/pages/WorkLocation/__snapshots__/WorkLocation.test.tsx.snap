// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkLocation Page Test Match Snapshot 1`] = `[Function]`;
